A72-open-telemetry-tracing.md

```diff
@@ -0,0 +1,680 @@
+plugin we will produce the following tracing information during an RPC lifecycle:
+
+At the client, on parent span:
+* If the RPC experienced name resolution delay, add an Event at the start of the
+  call with the name "Delayed name resolution complete" upon completion of the
+  name resolution process.
+* When the call is closed, set RPC status and end the parent span. gRPC status "OK" 
+  is recorded with status "OK", while other gRPC statuses are marked as "ERROR". 
+  Non-"OK" statuses include their code as a description, the same below.
+
+On attempt span:
+* When span is created, set the attribute with key `previous-rpc-attempts` and an 
+  integer value representing the count of previous attempts made for the RPC.
+* When span is created, set the attribute with key `transparent-retry` and a 
+  boolean value indicating whether the stream is undergoing a transparent retry.
+* If the RPC experienced load balancer pick delay, add an Event with the name 
+  "Delayed LB pick complete" upon creation of the stream on the transport. 
+* When the application sends an outbound message, add Event(s)
+  (it depends on implementation whether there is a single event or an additional 
+  separate event with name "Outbound message compressed" for compressed message size) 
+  with name "Outbound message sent" and the following attributes:
+  * key `message.event.type` with string value "SENT".
+  * key `message.message.id` with integer value of the seq no. The seq no. indicates
+    the order of the sent messages on the attempt (i.e., it starts at 0 and is 
+    incremented by 1 for each message sent), the same below.
+  * key named `message.event.size.uncompressed` if the message needs compression,
+    otherwise named `message.event.size`, with integer value of uncompressed
+    message size. The size is the total attempt message bytes without encryption,
+    not including grpc or transport framing bytes, the same below.
+  * If compression needed, add key `message.event.size.compressed` with integer 
+    value of compressed message size. If this is reported as a separate event in 
+    an implementation, the event name is "Outbound message compressed" and the 
+    order of the event must be after the first event that reports the message size.
+* When an inbound message has been received, add Event(s) (it depends on 
+  implementation whether there is a single event or an additional separate event
+  with name "Inbound message uncompressed" for uncompressed message size) with name 
+  "Inbound message read" and the following attributes:
+  * key `message.event.type` with string value "RECEIVED".
+  * key `message.message.id` with integer value of the seq no. The seq no. indicates
+    the order of the received messages on the attempt (i.e., it starts at 0 and is
+    incremented by 1 for each message received), the same below.
+  * key named `message.event.size.compressed` if the message needs decompression,
```

Comment:
@yashykt same here for inbound.

Reply:
On the inbound message, we know from the wire format whether the message was compressed, so this is fine.

But, it is illegal to have an attribute named `message.event.size` if there is another attribute `message.event.size.uncompressed` where  `message.event.size` is the namespace.